Source/Core/DolphinQt2/GameList/GameList.cpp

```diff
@@ -126,18 +165,169 @@ void GameList::OpenProperties()
 {
   QString game_id = GameFile(GetSelectedGame()).GetGameID();
   QString url = QStringLiteral("https://wiki.dolphin-emu.org/index.php?title=").append(game_id);
   QDesktopServices::openUrl(QUrl(url));
 }
 
+void GameList::CompressISO()
+{
+  const auto original_path = GetSelectedGame();
+  auto file = GameFile(original_path);
+
+  const bool compressed = (file.GetBlobType() == DiscIO::BlobType::GCZ);
+
+  if (!compressed && file.GetPlatformID() == DiscIO::Platform::WII_DISC)
+  {
+    QMessageBox wii_warning(this);
+    wii_warning.setIcon(QMessageBox::Warning);
+    wii_warning.setText(tr("Are you sure?"));
+    wii_warning.setInformativeText(
+        tr("Compressing a Wii disc image will irreversibly change the compressed copy by removing "
+           "padding data. Your disc image will still work."));
+    wii_warning.setStandardButtons(QMessageBox::Yes | QMessageBox::No);
+
+    if (wii_warning.exec() == QMessageBox::No)
+      return;
+  }
+
+  QString dst_path = QFileDialog::getSaveFileName(
+      this, compressed ? tr("Select where you want to save the decompressed image") :
+                         tr("Select where you want to save the compressed image"),
+      QFileInfo(GetSelectedGame())
+          .dir()
+          .absoluteFilePath(file.GetGameID())
+          .append(compressed ? QStringLiteral(".gcm") : QStringLiteral(".gcz")),
+      compressed ? tr("Uncompressed GC/Wii images (*.iso *.gcm") :
+                   tr("Compressed GC/Wii images (*.gcz)"));
+
+  if (dst_path.isEmpty())
+    return;
+
+  QProgressDialog progress_dialog(compressed ? tr("Decompressing...") : tr("Compressing..."),
+                                  tr("Abort"), 0, 100, this);
+  progress_dialog.setWindowModality(Qt::WindowModal);
+
+  bool good;
+
+  if (compressed)
+  {
+    good = DiscIO::DecompressBlobToFile(original_path.toStdString(), dst_path.toStdString(),
+                                        &CompressCB, &progress_dialog);
+  }
+  else
+  {
+    good = DiscIO::CompressFileToBlob(original_path.toStdString(), dst_path.toStdString(),
+                                      file.GetPlatformID() == DiscIO::Platform::WII_DISC ? 1 : 0,
+                                      16384, &CompressCB, &progress_dialog);
+  }
+
+  if (good)
+  {
+    QMessageBox(QMessageBox::Information, tr("Success!"), tr("Successfully compressed image."),
+                QMessageBox::Ok, this)
+        .exec();
+  }
+  else
+  {
+    QErrorMessage(this).showMessage(tr("Dolphin failed to complete the requested action."));
+  }
+}
+
+void GameList::InstallWAD()
+{
+  QMessageBox result_dialog(this);
+
+  const bool success = GameFile(GetSelectedGame()).Install();
+
+  result_dialog.setIcon(success ? QMessageBox::Information : QMessageBox::Critical);
+  result_dialog.setText(success ? tr("Succesfully installed WAD to NAND") :
+                                  tr("Failed to install WAD to NAND"));
+  result_dialog.exec();
+}
+
+void GameList::UninstallWAD()
```

Comment:
This should likely have a warning, just like in the WX implementation. I added it because I believe it is not obvious what gets removed from the NAND.

Unless you have a reason to use another wording, please keep the existing one to avoid people spending (unneeded) time on deciding on another wording.